check_gha_pinning/__init__.py

```diff
@@ -54,8 +79,8 @@ def _get_action_tags(action: str) -> tuple[dict[str, str], dict[str, list[str]]]
                     by_sha[sha].append(match.group(1))
         return by_tag, by_sha
     except subprocess.CalledProcessError as ex:
-        if ex.returncode >= 2:
-            raise _RefNotFoundError(f"repo {repo_url} not found")
+        if ex.returncode == 128:
```

Comment:
this seems fishy, https://git-scm.com/docs/git-ls-remote#Documentation/git-ls-remote.txt---exit-code says the exit code should be 0 or 2

Reply:
I think the documentation here is a bit misleading.

That return code (triggered by passing `--exit-code`) is *only* when no matching refs are found.

In this case (fetching all tags, not specific ones), 2 is only returned if the repository has no tags at all (I've just added a raise + handle for this)

128 is returned if the repository is not found at all (eg. `git ls-remote --tags --exit-code https://www.google.com`)

Reply:
gotcha, thanks!